internal/processor/parser/infra.go

```diff
@@ -31,109 +32,115 @@ func NewInfraParser(
 	}, nil
 }
 
-func (p InfraParser) Parse(envelope *events.Envelope) ([]metric.MetricPackage, error) {
+func (p InfraParser) Parse(envelope *loggregator_v2.Envelope) ([]metric.MetricPackage, error) {
 	metrics := []metric.MetricPackage{}
 
-	eventType := envelope.GetEventType()
-	if eventType != events.Envelope_ValueMetric && eventType != events.Envelope_CounterEvent {
+	switch envelope.GetMessage().(type) {
+	case *loggregator_v2.Envelope_Counter:
+		break
+	case *loggregator_v2.Envelope_Gauge:
+		if !util.IsContainerMetric(envelope) {
+			break
+		}
+		// if it's container metric, return error (we don't handle container metrics here)
+		// NOTE: `fallthrough` is not allowed in type switch
+		return metrics, fmt.Errorf("not an infra metric")
+	default:
 		return metrics, fmt.Errorf("not an infra metric")
 	}
 
 	host := parseHost(envelope)
-	name := getName(envelope)
 	tags := parseTags(envelope, p.Environment, p.DeploymentUUIDRegex, p.JobPartitionUUIDRegex)
 	tags = append(tags, p.CustomTags...)
 	tagsHash := util.HashTags(tags)
 
-	// create metricValues
-	metricValues := metric.MetricValue{}
-	metricValues.Host = host
-	metricValues.Tags = tags
-	value := getValue(envelope)
-	// NOTE: Value only has one point!!!!!!!!
-	metricValues.Points = append(metricValues.Points, metric.Point{
-		Timestamp: envelope.GetTimestamp() / int64(time.Second),
-		Value:     value,
-	})
-
-	var names []string
-	// Basic metric name
-	names = append(names, name)
-	// Legacy metric name
-	names = append(names, envelope.GetOrigin()+"."+name)
-	// BOSH alias metric name
-	if strings.HasPrefix(name, "bosh-hm-forwarder") {
-		names = append(names, strings.Replace(name, "bosh-hm-forwarder", "bosh.healthmonitor", 1))
-	}
-
-	// Create metric for each names with same values
-	for i := 0; i < len(names); i++ {
-		metrics = append(metrics, metric.MetricPackage{
-			MetricKey: &metric.MetricKey{
-				EventType: eventType,
-				Name:      names[i],
-				TagsHash:  tagsHash,
-			},
-			MetricValue: &metricValues,
+	for name, value := range getValues(envelope) {
+		// create metricValues
+		metricValues := metric.MetricValue{}
+		metricValues.Host = host
+		metricValues.Tags = tags
+		metricValues.Points = append(metricValues.Points, metric.Point{
```

Comment:
Based on my previous comment, could we investigate if having a list of `Point` is useful here?

Reply:
Yup, that sounds good, will take a look.

Reply:
So IIUC this is not possible. We could have list of points if we had multiple values for the same metric `name`, but what we have is multiple distinct metric names, each having a its own value.